application.py

```diff
@@ -143,11 +182,34 @@ def register_user():
         if not api_url:
             return jsonify({"error": "Internal server error"}), 500
 
-        # TODO: Set expiration date on verification token
-        verification_url = f"{api_url}/verify?jwt={create_access_token(email)}"
+        # Check if registration email already exists in backend_test.users
+        existing_user = User.query.filter_by(email_address=email).first()
+
+        if existing_user:
+            # CASE 1: Record with matching email exists and user is not verified (i.e., access_permissions is Null)
+            if existing_user.access_permissions is None:
+                # Resend verification email with updated token
+                verification_url = f"{api_url}/verify?jwt={create_access_token(email, expires_delta=timedelta(days=3))}"
+                send_verification_email(email, verification_url)
+
+                # Update the pending_registr_expiry_datetime to current datetime + 72 hours
+                # NOTE: Bad practice to use local time.
+                # TODO: Update all timestamping functionality to use UTC time
+                existing_user.pending_registr_expiry_datetime = datetime.now() + timedelta(days=3)
+
+                db.session.commit()
+
+                return jsonify({"message": "Verification email resent!"}), 200
+
+            # CASE 2: Record with matching email exists and user is already verified
+            return jsonify({"error": "Email is already registered to a verified account"}), 400
+
+        # Set expiration date on verification token
+        verification_url = f"{api_url}/verify?jwt={create_access_token(email, expires_delta=timedelta(days=3))}"
 
-        # TODO: Exception for user already registered
         user = User()
+        user.user_first_name = user_first_name
+        user.user_last_name = user_last_name
         user.email_address = email
         # TODO: Encrypt the given user password before storing it in database
```

Comment:
Please set `pending_registr_expiry_datetime` in the database when the user doesn't already exist in the users table. It looks like right now it's only getting set when the user already exists